portability-types-transfer/src/test/java/org/datatransferproject/types/transfer/auth/AuthDataSerializationTest.java

```diff
@@ -0,0 +1,90 @@
+package org.datatransferproject.types.transfer.auth;
+
+import static org.junit.Assert.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.ArrayList;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AuthDataSerializationTest {
+
+  private ObjectMapper objectMapper;
+
+  @Before
+  public void setUp() {
+    objectMapper = new ObjectMapper();
+    objectMapper.registerSubtypes(
+        CookiesAndUrlAuthData.class,
+        TokenAuthData.class,
+        TokensAndUrlAuthData.class,
+        TokenSecretAuthData.class);
+  }
+
+  @Test
+  public void verifyCookiesAndUrlAuthData() throws IOException {
+    final ArrayList<String> cookies = Lists.newArrayList("cookie_1", "cookie_2");
+    final String url = "https://www.example.com/auth";
+    final CookiesAndUrlAuthData authData = new CookiesAndUrlAuthData(cookies, url);
+
+    final String s = objectMapper.writeValueAsString(authData);
+    final AuthData readValue = objectMapper.readValue(s, AuthData.class);
+    assertTrue(
+        "The read AuthData should be an instance of CookiesAndUrlAuthData",
+        readValue instanceof CookiesAndUrlAuthData);
+    final CookiesAndUrlAuthData readAuthData = (CookiesAndUrlAuthData) readValue;
+    assertEquals("Expect cookies to be the same", cookies, readAuthData.getCookies());
```

Comment:
Eventually we may wan to add equals methods on these objects but this is a great start for now.